### PR TITLE
Move some macros to CHIPConfig.h

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

* Related #8006

#### Change overview

- This is PR contains trivial change to PR to move some macros from `src/app/InteractionModelEngine.h` to `src/lib/core/CHIPConfig.h` to make it configurable by `CHIPProjectConfig.h` with some renaming.
- Extracts discussions in #8006 as TODOs in this PR. They will be addressed after later updates to IM.

#### Testing
How was this tested? (at least one bullet point required)
* Build passed, no logic is changed in this PR.

### CODE REVIEW

File: src/lib/core/CHIPConfig.h [hunk 1]
```diff
@@ -2394,3 +2394,35 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_COMMISSIONING_HINT_INDEX_PRESS_RESET_SECONDS_WITH_POWER 10
 #define CHIP_COMMISSIONING_HINT_INDEX_PRESS_RESET_UNTIL_BLINK_WITH_POWER 11
 #endif
+
+#ifndef CHIP_IM_MAX_NUM_COMMAND_HANDLER
```
Comment (Damian-Nordic) [inline src/lib/core/CHIPConfig.h:2398]: nit: can you document these settings now or sometime in the future?

**Review by Damian-Nordic — APPROVED** [review]

File: src/lib/core/CHIPConfig.h (hunk 1 above)
Comment (erjiaqing) [reply src/lib/core/CHIPConfig.h:2421]: They should be included in this PR, updated, thanks.

Comment (Damian-Nordic) [reply src/lib/core/CHIPConfig.h:2421]: Thank you!

**Review by msandstedt — APPROVED** [review]
It is obviously much better for these to be configurable.  But this still will be insufficient for some targets where choosing this at compile time may be unworkable.

**Review by andy31415 — APPROVED** [review]

**Review by yunhanw-google — COMMENTED** [review]
thanks

**Pull request merged**